src/Api.Rest/Common/Utilities/OAuthHelper.cs

```diff
@@ -268,21 +296,17 @@ public static async Task<OAuthTokenCredential> GetAuthenticationInformationForDa
 		public static OAuthTokenCredential GetAuthenticationInformationForDatabaseUrl( string databaseUrl, string refreshToken = null, Func<OAuthRequest, OAuthResponse> requestCallback = null )
 		{
 			var instanceUrl = GetInstanceUrl( databaseUrl );
+			var discoveryCache = new DiscoveryCache( CreateAuthorityAsync( instanceUrl ).GetAwaiter().GetResult() );

```

Comment:
bitte `.ConfigureAwait( false )` ergaenzen, da sonst Deadlocks in den Anwendungen drohen (soweit ich weiss).

Reply:
an dieser Stelle waite ich doch garnicht oder verstehe ich das falsch?

Reply:
An dieser Stelle awaite ich doch garnicht. Was genau meinst du?

Reply:
Siehe hier https://devblogs.microsoft.com/dotnet/configureawait-faq/ insbesondere der Abschnitt "Avoiding Deadlocks". `await` ist syntaktischer Zucker fuer `GetAwaiter().GetResult()` (oder sehr aehnlich). Damit wartest du auch.

Reply:
De Aufruf hier ist synchron. Das ist aber noch viel schlimmer und wir sollten mal zusehen, dass wir das los werden.

Reply:
> `await` ist syntaktischer Zucker fuer `GetAwaiter().GetResult()` (oder sehr aehnlich).

Nein, das ist syntaktischer Zucker für .Wait mit etwas anderem Exception-Verhalten. Await ist was anderes und nur in diesem Kontext macht ConfigureAwait() Sinn.

Reply:
Ich glaube, dass es völlig sinnfrei ist, hier ConfigureAwait(false) zu ergänzen. Das hat nämlich hier überhaupt keine Auswirkungen. Es hätte eine Auswirkung, wenn wir den Task awaiten. Dann wäre bei der Rückkehr des Tasks der SynchronizationContext berücksichtigt worden. An dieser Stelle ist das aber nicht der Fall und so hat ConfigureAwait(...) keine Auswirkung. Ich mach es trotzdem hin, damit wir keine Grundsatzdiskussion beginnen.

@daniel-schroeder-zeiss : Es gibt eine asynchrone Variante dieser Methode (siehe `GetAuthenticationInformationForDatabaseUrlAsync`).

Reply:
Ohje, es lebe async/await. Macht einfach, wie ihr denkt.